crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -92,11 +92,9 @@ pub struct CheckpointStore {
         DBMap<ExecutionDigests, (CheckpointSequenceNumber, TxSequenceNumber)>,
 
     /// The mapping from checkpoint to transaction/effects contained within the checkpoint.
-    /// The second part of the key is the local sequence number if the transaction was
-    /// processed or Max(u64) / 2 + offset if not. It allows the authority to store and serve
-    /// checkpoints in a causal order that can be processed in order. (Note the set
-    /// of transactions in the checkpoint is global but not the order.)
-    pub checkpoint_contents: DBMap<(CheckpointSequenceNumber, TxSequenceNumber), ExecutionDigests>,
+    /// The checkpoint content should be causally ordered and is consistent among
+    /// all validators.
+    pub checkpoint_contents: DBMap<CheckpointSequenceNumber, CheckpointContents>,
```

Comment:
this isn't a new issue in this PR so no need to do anything, but this makes me wonder if we're going to run into any problems storing very large values in rocksdb - CheckpointContents could easily be megabytes at ~1000tps.

Reply:
Good point. I could also turn it back into a list of ExecutionDigest indexed by a local index (that's agreed upon)